### PR TITLE
Add DDL Export tests for Partition Functions/Schemes and Partitioned Tables/Indexes

### PR DESCRIPTION
**Pull request by sumitj824**
### Description

This commit adds tests to verify the correct DDL export behavior for various partitioning objects in Babelfish. The added tests cover the following scenarios to ensure accurate representation in the exported DDL:

- Partition Functions: Tests the export of partition functions, ensuring the function definition and parameters are correctly represented.

- Partition Schemes: Tests the export of partition schemes, ensuring the associated partition function and file group names are correctly represented.

- Partitioned Tables/Indexes: Tests the export of partitioned tables and indexes, ensuring the table/index definition, partition scheme, and partition column are correctly represented.

Signed-off-by: Sumit Jaiswal [sumiji@amazon.com](mailto:sumiji@amazon.com)
### Issues Resolved

Task: BABEL-5085

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/python/expected/pyodbc/ddl_partitions.out [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+CREATE PARTITION FUNCTION [SmallIntPartitionFunction](smallint) AS RANGE RIGHT FOR VALUES (-32768, 0, 32767)
+GO
+
+CREATE PARTITION FUNCTION [SmallMoneyPartitionFunction](smallmoney) AS RANGE RIGHT FOR VALUES (2.5679, 3.5679, 4.9100)
+GO
+
+CREATE PARTITION FUNCTION [SqlVariantPartitionFunction](sql_variant) AS RANGE RIGHT FOR VALUES (N'abc  ', N'Apple', N'Banana', N'Some text', N'xyz  ')
+GO
+
+CREATE PARTITION FUNCTION [TinyIntPartitionFunction](tinyint) AS RANGE RIGHT FOR VALUES (0x00, 0x80, 0xFF)
```
Comment (shah-nirmit) [inline test/python/expected/pyodbc/ddl_partitions.out:67]: Question: You mentioned `sys.tinyint` in your definition , while here it reflects `tinyint` . Is this expected?

File: test/python/SMO_script.ps1 [hunk 2]
```diff
@@ -59,15 +83,12 @@ else
     $SubObjects += $db.Tables.Indexes
     $SubObjects += $db.Tables.Triggers
     $SubObjects += $db.Users
+
+    # Scripting standard database objects.
     foreach ($CurrentObject in $Objects)
     {
         if ($CurrentObject.schema -ne $schm -and $CurrentObject.schema -ne $dtb -and $CurrentObject.schema -ne $null -and -not $CurrentObject.IsSystemObject )
         {
-            $Scripter = New-Object ('Microsoft.SqlServer.Management.Smo.Scripter') ($SmoServer)
```
Comment (shah-nirmit) [inline test/python/SMO_script.ps1:66]: Thanks for moving this above.

**Review by shah-nirmit — APPROVED** [review]

**Comment by shah-nirmit** [comment]
The DDL export test seem to [fail](https://github.com/babelfish-for-postgresql/babelfish_extensions/actions/runs/10099071590/job/27927464008?pr=2782) with some diff. Can you update the expected files please

File: test/python/expected/pyodbc/ddl_partitions.out (hunk 1 above)
Comment (sumitj824) [reply test/python/expected/pyodbc/ddl_partitions.out:67]: yes, input file was supposed to test type with explicit schema should also work.

**Review by sumitj824 — COMMENTED** [review]
Test failure will get resolved after rebasing the PR, once the System views changes are merged.

**Review by shah-nirmit — APPROVED** [review]

**Review by KushaalShroff — APPROVED** [review]

File: test/python/expected/pyodbc/ddl_partitions.out [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+CREATE PARTITION SCHEME [VarCharPartitionScheme] AS PARTITION [VarCharPartitionFunction] TO ([PRIMARY], [PRIMARY], [PRIMARY], [PRIMARY], [PRIMARY], [PRIMARY], [PRIMARY])
+GO
+
+GO
+
+GO
+
+SET ANSI_NULLS ON
+SET QUOTED_IDENTIFIER ON
+CREATE TABLE [dbo].[partition_vu_prepare_bigint_table](
```
Comment (Deepesh125) [inline test/python/expected/pyodbc/ddl_partitions.out:160]: should we have test that uses UDT?

**Pull request merged**